core/src/main/scala-3/pureconfig/generic/derivation/EnumConfigWriterDerivation.scala

```diff
@@ -0,0 +1,43 @@
+package pureconfig
+package generic
+package derivation
+
+import scala.compiletime.{constValue, erasedValue, error, summonInline}
+import scala.deriving.Mirror
+
+import com.typesafe.config.{ConfigValue, ConfigValueFactory}
+
+import pureconfig.error.{CannotConvert, ConfigReaderFailures}
+import pureconfig.generic.derivation.Utils.*
+
+type EnumConfigWriter[A] = EnumConfigWriterDerivation.Default.EnumConfigWriter[A]
+
+trait EnumConfigWriterDerivation(transformName: String => String) {
+
+  trait EnumConfigWriter[A] extends ConfigWriter[A]
+
+  object EnumConfigWriter {
+    inline def derived[A](using m: Mirror.SumOf[A]): EnumConfigWriter[A] =
+      new EnumConfigWriter[A] {
+        assertIsEnum[m.MirroredElemTypes]
+        val labels = transformedLabels[A](transformName)
```

Comment:
Can you please make this a `Vector`, to avoid a O(n) label search on every write?

Reply:
Yeah, certainly! I forgot to port it 😄